test/integration/conftest.py

```diff
@@ -188,6 +202,16 @@ def build_relation_charm(relation_charm_dir: pathlib.Path) -> Generator[Callable
     yield from _build_charm(relation_charm_dir, 'test-relation_amd64.charm')
 
 
+@pytest.fixture(scope='session')
+def build_hookcmds_charm(hookcmds_charm_dir: pathlib.Path) -> Generator[Callable[[], str]]:
+    """Build the test_hookcmds charm and provide the artefact path.
+
+    Starts building the test-hookcmds charm early.
+    Call the fixture value to get the built charm file path.
+    """
+    yield from _build_charm(hookcmds_charm_dir, 'test-hookcmds_amd64.charm')
```

Comment:
Can the architecture not be hardcoded?

Reply:
This is also just following the existing pattern. Maybe best to split out into a separate PR?

Reply:
Got it. Sure, we can ignore it here.